torchvision/ops/dice_loss.py

```diff
@@ -0,0 +1,71 @@
+import torch
+import torch.nn.functional as F
+
+from ..utils import _log_api_usage_once
+
+
+def dice_loss(inputs: torch.Tensor, targets: torch.Tensor, reduction: str = "none", eps: float = 1e-7) -> torch.Tensor:
+    r"""Criterion that computes Sørensen-Dice Coefficient loss.
+
+    We compute the Sørensen-Dice Coefficient as follows:
+
+    .. math::
+
+        \text{Dice\_Loss}(X, Y) = 1 - \frac{2 |X \cap Y|}{|X| + |Y|}
+
+    Where:
+       - :math:`X` expects to be the scores of each class.
+       - :math:`Y` expects to be the one hot tensor with the class labels.
+
+    the loss, is finally computed as:
+
+    .. math::
+
+        \text{loss}(x, class) = 1 - \text{Dice}(x, class)
+
+    Args:
+        inputs: (Tensor): A float tensor with rank >= 2 and shape (B, C, N1, .... NK)
```

Comment:
As suggested by nicolas. Rename C to `num_classes`

Reply:
I quickly checked and yes we are following putting `num_boxes`, `num_masks` as first channel in Tensor and then followed by C H W. 
In general C in torchvision code is used for channels.

You can probably refer functions like 
`masks_to_boxes`. `box_iou`, etc methods
https://github.com/pytorch/vision/blob/main/torchvision/ops/boxes.py#L386